vrx_gz/worlds/follow_path_task.sdf

```diff
@@ -484,7 +484,21 @@
       <release_topic>/vrx/release</release_topic>
 
       <course_name>short_navigation_course_0</course_name>
-      <obstacle_penalty>10.0</obstacle_penalty>
+      <points_per_gate_crossed>10</points_per_gate_crossed>
+      <obstacle_penalty>3.0</obstacle_penalty>
+      <gate_bonus>
+        <gate>1</gate>
+        <gate>2</gate>
+        <gate>4</gate>
+        <gate>7</gate>
+        <gate>11</gate>
+        <gate>16</gate>
+        <gate>22</gate>
+        <gate>29</gate>
+        <gate>37</gate>
+        <gate>46</gate>
+        <gate>56</gate>
```

Comment:
Using these bonus values, the points awarded for going through consecutive gates would be:
| Length of Run  | Added Pts | Total Bonus | 
| ----------- | ----------- |-|
| 1   | 0 |  0    |
| 2   | 1 | 1   |
| 3   | 2 | 3   |
| 4   | 4 | 7   |
| 5   | 7 | 14   |
| 6   | 11 | 25   |
| 7   | 16 | 41   |
| 8   | 22 | 63   |
| 9   | 29 | 92   |
| 10 | 37 |  129 |
| 11 | 46 | 175  |
| 12 | 56 |  231 |

This seems to be the summation of the [lazy caterer's sequence](https://en.wikipedia.org/wiki/Lazy_caterer%27s_sequence#:~:text=1%2C%202%2C%204%2C%207,gives%20the%20lazy%20caterer's%20sequence.)? Was that your intention?

If you want to use summation to make the total bonus equal to the square of the length of the run, I think you can just set the individual values to the odd integers:
| Length of Run   | Added Pts | Total Bonus | 
| ----------- | ----------- |-|
| 1   | 1   | 1 |
| 2   | 3   | 4 |
| 3   | 5   | 9 |
| 4   | 7   | 16 |
| 5   | 9   | 25 |
| ...   | etc.   |

If the bonus doesn't start until 2 consecutive gates have been passed, then it would be:
| Length of Run   | Added Pts | Total Bonus |
| ----------- | ----------- |-|
| 1   | 0   | 0 |
| 2   | 4   | 4 |
| 3   | 5   | 9 |
| 4   | 7   | 16 |
| 5   | 9   | 25 |
| ...   | etc.   |

Reply:
That was my intention but I didn't know the name of that sequence. In any case, I updated the code to your suggested bonus sequence to make it exactly equal to the square of the length of the run.

Reply:
@caguero OK, thanks. I'm not sure squares are best--I was just confused about what the pattern was.

I think we should revisit the impact of these bonuses when we next meet. It occurred to me that there are some strange effects of the squares bonus approach I suggested. One of them is that a team that only misses one gate of a 9 gate run will do much better if it's the first or last gate than if it's a gate somewhere in the middle. Another, maybe more concerning problem, is that the bonus can sometimes allow a team to score higher than another team that passed through more gates correctly.

I calculated the scores for all possible runs over a 9-gate course and the results convinced me that @j-herman 's original suggestion of adding a bonus equal to the current length of the run avoids the second problem for the kinds of courses we're using.

Additionally, the results will be almost the same if we simply break ties by the length of the longest consecutive run. So that might be even better. I'll send you both the numbers and you can see what you think.